taskgraph/Task.py

```diff
@@ -105,6 +109,103 @@ def _initialize_logging_to_queue(logging_queue):
+
+    Returns:
+        None.
+
+    """
+    sql_create_projects_table_script = (
+        """
+        CREATE TABLE taskgraph_data (
+            task_reexecution_hash TEXT NOT NULL,
+            target_path_stats BLOB NOT NULL,
+            result BLOB NOT NULL,
+            PRIMARY KEY (task_reexecution_hash)
+        );
+        CREATE TABLE global_variables (
+            key TEXT NOT NULL,
+            value BLOB,
+            PRIMARY KEY (key)
+        );
+        """)
+
+    table_valid = True
+    expected_table_column_name_map = {
+        'taskgraph_data': [
+            'task_reexecution_hash', 'target_path_stats', 'result'],
+        'global_variables': ['key', 'value']}
+    if os.path.exists(taskgraph_database_path):
+        try:
+            # check that the tables exist and the column names are as expected
+            for expected_table_name in expected_table_column_name_map:
+                table_result = _execute_sqlite(
+                    '''
+                    SELECT name
+                    FROM sqlite_master
+                    WHERE type='table' AND name=?
+                    ''', taskgraph_database_path,
+                    argument_list=[expected_table_name],
+                    mode='read_only', execute='execute', fetch='all')
+                if not table_result:
+                    raise ValueError(f'missing table {expected_table_name}')
+
+                # this query returns a list of results of the form
+                # [(0, 'task_reexecution_hash', 'TEXT', 1, None, 1), ... ]
+                # we'll just check that the header names are the same, no
+                # need to be super aggressive, also need to construct the
+                # PRAGMA string directly since it doesn't take arguments
+                table_info_result = _execute_sqlite(
+                    f'PRAGMA table_info({expected_table_name})',
+                    taskgraph_database_path, mode='read_only',
+                    execute='execute', fetch='all')
+
+                expected_column_names = expected_table_column_name_map[
+                    expected_table_name]
+                header_count = 0
+                for header_line in table_info_result:
+                    column_name = header_line[1]
+                    if column_name not in expected_column_names:
+                        raise ValueError(
+                            f'expected {column_name} in table '
+                            f'{expected_table_name} but not found')
+                    header_count += 1
+                if header_count < len(expected_column_names):
+                    raise ValueError(
+                        f'found only {header_count} of an expected '
+                        f'{len(expected_column_names)} columns in table '
+                        f'{expected_table_name}')
+                if not table_info_result:
+                    raise ValueError(f'missing table {expected_table_name}')
+        except Exception:

```

Comment:
Does it make sense to only except `ValueError` here? I'm not sure. I could also see wanting to bail out and remove the database if anything else unexpected happens too.

Reply:
I see what you're saying. Conceptually I'm not explicitly guarding against `ValueError`s here, but really anything that goes wrong while checking the status of the database. So lemme make something up and say the "database" is actually some corrupt binary file, I don't know if that would throw and OSError or RuntimeError or what, but I sure wouldn't want it to be the database. I put a comment there describing why just `Exception` for clarity in 210a8fc817179b181ed7c755f9bf7cc087179e58.